### DISCUSSION TITLE
Making overlays fullscreen

### DISCUSSION THREAD

**Discussion by salman-farooq-sh** [helix-editor/helix#11780] [Ideas]
I have been using this on my own for a while and it has been great having maximum file preview area. In case anybody wants to try it:

https://github.com/salman-farooq-sh/helix
https://github.com/salman-farooq-sh/helix-pkgbuild

![image](https://github.com/user-attachments/assets/f3162dc1-350c-4fbc-bf18-f6b8dee135d5)

**Comment by rahil627** [comment]
posting a screenshot may help!

**Reply by salman-farooq-sh** [reply]
Updated!

**Comment by rahil627** [comment]
ahhhh..! hmm, it's a very small nitpick, but maybe more useful for people with small terminal windows..?? i personally use a page-like size and don't even get to see the preview-side... i like the default centered design, but maybe there can be less padding?

**Reply by salman-farooq-sh** [reply]
It's not a small nitpick for me because I have to use a large font size which makes my terminal width and height relatively small even when the terminal is full-screen.

I think this is a better utilization of space in the landscape orientation (i.e. when width > height) nonetheless.

For the people that do use the previews, this is a free upgrade in my opinion at no special cost I guess. The content behind the overlays isn't properly readable anyways so why not make use of that free real estate? :)

I do understand this is a major component design change and it might need much more discussion to happen but my vote is for it to happen.

If you don't use the previews, then I guess this will neither trouble you nor benefit you much.

**Reply by rahil627** [reply]
yeah... i think you're right: it doesn't hurt anyone to maximize it. it's only padded for aesthetics. who knows, maybe then i'll finally be able to see the preview! 😅

**Reply by salman-farooq-sh** [reply]
Well, you can try my fork! If you are on archlinux it will be as simple as `makepkg -si` in the pkgbuild repo.

**Comment by salman-farooq-sh** [comment]
@the-mikedavis ping

**Comment by nik-rev** [comment]
I think this is a really good idea. I personally never use the outside space so using that for bigger file previewes would be golden

I assume there would need to be a new config option for this

**Reply by salman-farooq-sh** [reply]
Personally, I would like this to be the default so as to not bloat the config. But I can make this config-able too if the popular agreement is on that.

**Reply by nik-rev** [reply]
> Personally, I would like this to be the default so as to not bloat the config. But I can make this config-able too if the popular agreement is on that.

Ok, so somewhere in the codebase we are multiplying by `0.9` or something along those lines to actually reduce the picker's height and width. I've been ripgrepping and searching for a while now and I can't find where I saw it

I even remember myself changing this value and the picker becoming full height and width. But I forgot where it is

If you find it will be a simple 1 line change to make a PR for this.

Check out the `render` method in `helix-term/src/ui/picker.rs:935`. I think that will be helpful

**Reply by salman-farooq-sh** [reply]
I tried opening a PR for this before but I couldn't for some reason at that time. The required code change was already linked in the original post. Here it is now https://github.com/helix-editor/helix/pull/12311.

**Reply by debajyoti1990** [reply]
> Personally, I would like this to be the default so as to not bloat the config. But I can make this config-able too if the popular agreement is on that.

I would disagree on this being default behavior. Current overlay window size would be enough for most of us.

**Reply by salman-farooq-sh** [reply]
What about small terminal sizes or needing large font sizes which some people don't have a choice about?

How will this negatively impact any usability? I can only see benefits.

**Reply by debajyoti1990** [reply]
> What about small terminal sizes or needing large font sizes which some people don't have a choice about?
> 
> How will this negatively impact any usability? I can only see benefits.

Overlay window smaller than full screen aesthetically looks better IMO and for the people who have the limitations mentioned, I fail to see any significant improvement because of full screen overlay.

**Reply by salman-farooq-sh** [reply]
So usability (not aesthetics, about which I agree with you) is not affected with this for you, correct?

Have you tried working on small terminal sizes or with large fonts both of which make the terminal have a reduced column and row count? It is much better in those situations that you are able to use the most amount of screen space, and not have even smaller pickers and/or previews.
